src/ship_cmd.cpp

```diff
@@ -912,6 +912,10 @@ CommandCost CmdBuildShip(DoCommandFlags flags, TileIndex tile, const Engine *e,
 		v->y_pos = y;
 		v->z_pos = GetSlopePixelZ(x, y);
 
+		v->direction = DiagDirToDir(GetShipDepotDirection(tile));
+		v->rotation = v->direction;
+		v->rotation_x_pos = v->x_pos;
+		v->rotation_y_pos = v->y_pos;
```

Comment:
Why do these need to be set?

Reply:
Because `v->UpdateDeltaXY()` will be looking into them next line.

Reply:
If someone has to ask the question, it often means a comment is appropriate.

```c++
/* UpdateDeltaXY() requires rotation members to be initialised as well. */
```

Reply:
If `v->rotation` is set equal to `v->direction`, the other member variables are not even used.

In the changes in 11a8b71 only `direction` and `rotation` got another value during initialisation, so why would any other values now need to initialised differently?

Reply:
It works without
```
v->rotation_x_pos = v->x_pos;
v->rotation_y_pos = v->y_pos;
```
But I was trying to follow the convention used in `SLV_SHIP_ROTATION` which initializes these variables in afterload.